core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -474,17 +499,16 @@ default Builder retryStrategy(RetryMode retryMode) {
         Builder retryStrategy(RetryStrategy retryStrategy);
 
         /**
-         * Configure the retry strategy that should be used when handling failure cases.
+         * Configure a new default retry strategy that should be used when handling failure cases using the given consumer for it.
```

Comment:
We should have the same wording for all three methods - I can see they're mostly similar. To modify:
- each should say that the retryMode methods are mutually exclusive
- each should say that they override policy
- this method says `new default` and the others do not - let's consistently use one or the other. 
I'm a bit nitpicky here but I think this topic requires clarity!

Reply:
> We should have the same wording for all three methods - I can see they're mostly similar. To modify:

Can you please suggest a new wording? That would make it a lot easier than guessing what you're thinking about, e.g., 

> each should say that the retryMode methods are mutually exclusive

There's only one method using retry model, so there are not many that can be *mutually exclusive*.

> each should say that they override policy

As in retry policy? Those do not override policies, I guess we can but we don't at the moment.

> this method says new default and the others do not - let's consistently use one or the other.

The idea is that this method will create a *new* default retry strategy (as given by `AwsRetryStrategy.defaultRetryStrategy()`) and pass that to the `mutator`. If we had access to `aws-core` it would be implemented as 

```java
    default Builder retryStrategy(Consumer<RetryStrategy.Builder<?, ?>> mutator) {
        RetryStrategy.Builder<?, ?> defaultBuilder = AwsRetryStrategy.defaultRetryStrategy().toBuilder();
        configurator.accept(defaultBuilder);
        return retryStrategy(defaultBuilder.build());
    }
```

Reply:
Thanks for the updates. 

Sorry, I meant to say that the `retryStrategy` methods are mutually exclusive settings, not retryMode. That was confusing. If we can include a reference to that, I'd appreciate it. (Nit)
`Note that retryStrategy options are mutually exclusive`?

Disregard the comment about policy, I think that wording is gone anyway.